libs/utils/decorate.js

```diff
@@ -228,12 +228,73 @@ function getVideoAttrs(hash, dataset) {
   return `${globalAttrs} controls`;
 }
 
+export function syncPausePlayIcon(video) {
+  const playIcon = video.nextElementSibling?.querySelector('.play-icon');
+  const pauseIcon = video.nextElementSibling?.querySelector('.pause-icon');
+  if (video.paused || video.ended) {
+    playIcon?.classList.remove('hidden');
+    pauseIcon?.classList.add('hidden');
+  } else {
+    pauseIcon?.classList.remove('hidden');
+    playIcon?.classList.add('hidden');
```

Comment:
Could we not apply a class to their parent and then show/hide the appropriate child from CSS based on it?

Reply:
isn't it implement in the same way now?
I am showing/hiding the child that i get from the video's sibling based on css.
could you elaborate a bit more if i have misunderstood this?

Reply:
What I mean by this is - instead of toggling the children classes
```
<div class="wrapper">
  <img class="play" ... />
  <img class="pause hidden" ... />
</div>
```
you could toggle the parent class and then show/hide the appropriate element from CSS:
```
<div class="wrapper is-playing"> <!-- or 'is-paused' -->
  <img class="play" ... />
  <img class="pause" ... />
</div>
```
That would mean a single class update instead of 2.